sklearn/manifold/_barnes_hut_tsne.pyx

```diff
@@ -133,7 +133,7 @@ cdef float compute_gradient_positive(float[:] val_P,
             for ax in range(n_dimensions):
                 buff[ax] = pos_reference[i, ax] - pos_reference[j, ax]
                 dij += buff[ax] * buff[ax]
-            qij = (((1.0 + dij) / dof) ** exponent)
+            qij = ((1.0 + dij / dof) ** exponent)
```

Comment:
Is this computing the student t or the derivative?

Reply:
This is actually used in both calculations: the cost and the gradient. 
The gradient is (y_i - y_j) p_ij (1 + (||y_i - y_j||^2)/dof)**exponent 
Final qij is just (1 + (||y_i - y_j||^2)/dof)**exponent  normalized so that sum of all qij is 1, and the cost is computed in line 39, which is KL(p||q)
In that line qij is just storing (1 + (||y_i - y_j||^2)/dof)**exponent which will be used in both the gradient and the cost